org.code.toboggan.ui/src/org/code/toboggan/ui/dialogs/extensions/DialogWSEvent.java

```diff
@@ -14,8 +20,25 @@
 import org.eclipse.swt.widgets.Display;
 import org.eclipse.swt.widgets.Shell;
 
+import clientcore.dataMgmt.SessionStorage;
+
 public class DialogWSEvent implements IWSEvent {
 	private Logger logger = LogManager.getLogger(this.getClass());
+	
+	public DialogWSEvent(){
+		CoreActivator.getSessionStorage().addPropertyChangeListener(new PropertyChangeListener() {
+			
+			@Override
+			public void propertyChange(PropertyChangeEvent evt) {
+				if (evt.getPropertyName().equals(SessionStorage.USERNAME)){
+					Display.getDefault().asyncExec(() -> {
+						Shell shell = Display.getDefault().getActiveShell();
+						new ConfirmSubscribeDialog(shell).open();
+					});
+				}				
+			}
+		});
```

Comment:
That may be true that it's websocket related, but it's listening for the change in SessionStorage which exists in the Core plugin. I'm not suggesting putting this code in the Core plugin, but rather make a property change listener that triggers an extension point called maybe "storageListener" or something to that effect. Take a look at the notifyWSEventListeners() in WSService for an example. That's how this DialogWSEvent class is triggered, and this is still able to exist in the UI plugin.